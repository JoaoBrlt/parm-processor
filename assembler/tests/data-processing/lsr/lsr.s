// Shift by a bit.
// 11111111 11111111 11111111 01010101 >> 00001 =
// 01111111 11111111 11111111 10101010
MOVS r0, #1
MOVS r1, #170
MVNS r1, r1
LSRS r1, r0

// Shift by seven bits.
// 11111111 11111111 11111111 01010101 >> 00111 =
// 00000001 11111111 11111111 11111110
MOVS r2, #7
MOVS r3, #170
MVNS r3, r3
LSRS r3, r2

// Shift by a byte.
// 11111111 11111111 11111111 11111111 >> 01000 =
// 00000000 11111111 11111111 11111111
MOVS r4, #8
MOVS r5, #0
MVNS r5, r5
LSRS r5, r4

// Shift by three bytes.
// 11111111 11111111 11111111 00110011 >> 11000 =
// 00000000 00000000 00000000 11111111
MOVS r6, #24
MOVS r7, #204
MVNS r7, r7
LSRS r7, r6

// Shift by maximum.
// 11111111 11111111 11111111 00000000 >> 11111 =
// 00000000 00000000 00000000 00000001
MOVS r0, #31
MOVS r1, #255
MVNS r1, r1
LSRS r1, r0

// Shift by over maximum.
// 11111111 11111111 11111111 00000000 >> 00000 =
// 11111111 11111111 11111111 00000000
MOVS r2, #32
MOVS r3, #255
MVNS r3, r3
LSRS r3, r2

// Shift itself by a bit.
// ...10000001 >> 00001 = ...01000000
MOVS r4, #129
LSRS r4, r4
