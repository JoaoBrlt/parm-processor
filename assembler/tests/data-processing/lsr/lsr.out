v2.0 raw
2001 21aa 43c9 40c1 2207 23aa 43db 40d3 2408 2500 43ed 40e5 2618 27cc 43ff 40f7 201f 21ff 43c9 40c1 2220 23ff 43db 40d3 2481 40e4 defd
